docs/examples/mm_structured_outputs.md

```diff
@@ -0,0 +1,431 @@
+        **JUDGE_PARAMS,
+    ),
+).collect()
+
+print(f"Judged {df_judged.count_rows()} failure rows")
+```
+
+### Interpreting Judge Feedback
+
+Use the judge’s `attribution` signal to quickly separate **question issues** (ambiguous prompt/choices) from **image understanding issues** (missed labels, small text, visual ambiguity). For more on these failure modes and what we observed at scale, see the accompanying blog post: [Multimodal Structured Outputs: Evaluating VLM Image Understanding at Scale](https://www.daft.ai/blog/multimodal-structured-outputs-evaluating-vlm-image-understanding-at-scale).
+
+Inspect a few judged failures:
+
+```python
+from daft.functions import unnest
+
+df_judged.select(
+    "quadrant",
+    "user",
+    "image",
+    "answer",
+    col("result")["choice"].alias("with_image"),
+    col("result_no_image")["choice"].alias("without_image"),
+    unnest(col("judge_response")),
+).show(3)
+```
+
+Sanity-check that we ran every stage:
+
+```python
+print(f"Accuracy (with image):    {accuracy:.1%}")
+print(f"Accuracy (without image): {accuracy_no_image:.1%}")
+print(f"Delta:                   {accuracy - accuracy_no_image:+.1%}")
+
+df_classified.groupby("quadrant").count().show()
+
+print(f"Judge rows: {df_judged.count_rows()}")
+```
+
+## 7. Scale with Daft Cloud
+
+**Everything above runs locally on 50 rows.**
+
+But The Cauldron contains **millions of rows across 50 subsets**. To run this evaluation at scale with strong consistent performance we can scale on [Daft Cloud](https://daft.ai/cloud). The python script version of this notebook is available in the [daft-examples](https://github.com/Eventual-Inc/daft-examples) repo in the [use_cases/image_understanding_eval](https://github.com/Eventual-Inc/daft-examples/tree/main/use_cases/image_understanding_eval) directory.
+
+👉 [**Sign up for early access**](https://daft.ai/cloud) | [**Book a demo**](https://www.daft.ai/demo)
+
+### Take this one step further with LLM-as-a-judge
+
+For a daft cloud ready-to-run script with a bonus LLM-as-a-judge section, check out [`eval_image_understanding.py`](https://github.com/Eventual-Inc/daft-examples/blob/main/use_cases/image_understanding_eval/eval_image_understanding.py) in the [daft-examples](https://github.com/Eventual-Inc/daft-examples) repo.
+
+## 8. Conclusion
+
+In this notebook, we built a small pipeline to evaluate Qwen3-VL's image understanding:
+
+1. **Structured Outputs**: Used Pydantic models to enforce consistent responses
+2. **Ablation Study**: Isolated image understanding from general reasoning
+3. **Quadrant Analysis**: Classified results into actionable categories
+4. **LLM-as-a-Judge**: Diagnosed failures on the most informative subsets ("Image Hurt" + "Both Incorrect")
+
+### Next Steps
+
+**Multi-Dataset Evaluation**: Extend across all 50 Cauldron subsets
+
+```python
+subsets = ["ai2d", "chartqa", "docvqa", "infographicvqa", ...]
+for subset in subsets:
+    df = run_full_pipeline(subset, MODEL_ID)
```

Comment:
I think it'll be helpful to explain what this function is supposed to do here because this is the first time you mentioned this particular function.

Plus this is still in a copy and pasteable code block even though it's pseudocode.